### PR TITLE
[TPC] Checkstyle issues resolved.

### PR DESCRIPTION
**Pull request by pveentjer**
All the checkstyle issues have been resolved and checkstyle is now enabled again.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/TPCOperationExecutor.java [hunk 1]
```diff
@@ -469,7 +469,8 @@ public int toPartitionThreadIndex(int partitionId) {
     @Override
     public void start() {
         if (logger.isFineEnabled()) {
-            logger.fine("Starting  " + genericThreads.length + " generic threads (" + priorityThreadCount + " dedicated for priority tasks)");
+            logger.fine("Starting  " + genericThreads.length + " generic threads (" + priorityThreadCount
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/TPCOperationExecutor.java:472]:
```suggestion
            logger.fine("Starting " + genericThreads.length + " generic threads (" + priorityThreadCount
```

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Eventloop.java [hunk 2]
```diff
@@ -58,7 +58,7 @@
  *     <li>Tasks from some asynchronous eventing system that interacts with I/O. </li>>
  * </ol>
  */
-@SuppressWarnings({"checkstyle:VisibilityModifier", "rawtypes"})
+@SuppressWarnings({"checkstyle:VisibilityModifier", "checkstyle:declarationorder", "rawtypes"})
```
Comment (mdumandag) [inline hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Eventloop.java:61]: Just a question: Are we going to get rid of (maybe some?) warnings before merging to master, or do we plan to leave them with these suppressions?

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/iobuffer/ConcurrentIOBufferAllocator.java [hunk 3]
```diff
@@ -29,13 +29,17 @@
...
 public class ConcurrentIOBufferAllocator implements IOBufferAllocator {
 
-    private static final AtomicLong newAllocations = new AtomicLong(0);
-    private static final AtomicLong pooledAllocations = new AtomicLong(0);
-    private static final AtomicLong allocateCalls = new AtomicLong();
-    private static final AtomicLong releaseCalls = new AtomicLong();
-    private final static ThreadLocal<Pool> POOL = new ThreadLocal<>();
+    // this is for debugging. We do not want to track these calls normally
+    private static final AtomicLong NEW_ALLOCATIONS = new AtomicLong(0);
+    private static final AtomicLong POOLED_ALLOCATIONS = new AtomicLong(0);
```
Comment (mdumandag) [inline hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/iobuffer/ConcurrentIOBufferAllocator.java:37]:
```suggestion
    private static final AtomicLong NEW_ALLOCATIONS = new AtomicLong();
    private static final AtomicLong POOLED_ALLOCATIONS = new AtomicLong();
```
minor: just to be consistent with other two

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java [hunk 4]
```diff
@@ -144,6 +144,8 @@ private ClientMessage() {
 
     }
 
+
+
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java:148]: minor: unnecessary newlines

**Review by Fly-Style — APPROVED** [review]
LGTM, I'll send a follow-up PR with few enhancements

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Eventloop.java (hunk 2 above)
Comment (pveentjer) [reply hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Eventloop.java:61]: With suppression. This is a nonsense checkstyle constraint on code. You should never group based on the access modifier but on what belongs to each other.

Comment (pveentjer) [reply hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Eventloop.java:61]: This whole section needs to be improved anyway due to potential false sharing issues. THe (mutable) state that is accessed by the eventloop thread should not be on the same cache lines as state accessed by other threads. So I need to add padding.

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java (hunk 4 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java:148]: fixed

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/iobuffer/ConcurrentIOBufferAllocator.java (hunk 3 above)
Comment (pveentjer) [reply hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/iobuffer/ConcurrentIOBufferAllocator.java:37]: Fixed

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/TPCOperationExecutor.java (hunk 1 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/TPCOperationExecutor.java]: Fixed

**Review by mdumandag — APPROVED** [review]

**Pull request merged**